### DISCUSSION TITLE
restreamer not waiting on ffmpeg to start

### DISCUSSION THREAD

**Discussion by AaronNGray** [datarhei/restreamer#810] [General]
On Video Setup and RTMP Server. On  hitting next I get a red warning sign and :-

    No live stream was detected.Please check the software that sends the stream

```
ffmpeg -re -i big_buck_bunny_720p_h264.mov -vcodec libx264 -acodec aac -preset ultrafast -tune zerolatency -f flv rtmp://<my-ip>/live/bbb/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream/1234
```
is giving a socket error :-

    Connection to tcp://<my-ip>:1935 failed: Connection refused
    [rtmp @ 0x63017c2a6080] Cannot open connection tcp://<my-ip>:1935
    rtmp://<my-ip>/live/bbb/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream/1234: Connection refused

Basically either one wants the other to have connected first and its loggerheads

**Comment by AaronNGray** [comment]
`-listen 100` does not seem to do anything :-

    ffmpeg -re -i big_buck_bunny_720p_h264.mov -vcodec libx264 -acodec aac -preset ultrafast -tune zerolatency -f flv -listen 100 rtmp://<my-ip>/live/bbb/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream/1234

**Comment by AaronNGray** [comment]
The following will stream to localhost :-

    ffmpeg -re -i "big_buck_bunny_720p_h264.mov" -pix_fmt yuv420p -vsync 1 -threads 0 -vcodec libx264 -r 30 -g 60 -sc_threshold 0 -b:v 512k -bufsize 640k -maxrate 640k -preset veryfast -profile:v baseline -tune film -acodec aac -b:a 128k -ac 2 -ar 48000 -af "aresample=async=1:min_hard_comp=0.100000:first_pts=0" -bsf:v h264_mp4toannexb -f mpegts "rtmp://localhost/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream live=1"

but :-

    ffmpeg -re -i "big_buck_bunny_720p_h264.mov" -pix_fmt yuv420p -vsync 1 -threads 0 -vcodec libx264 -r 30 -g 60 -sc_threshold 0 -b:v 512k -bufsize 640k -maxrate 640k -preset veryfast -profile:v baseline -tune film -acodec aac -b:a 128k -ac 2 -ar 48000 -af "aresample=async=1:min_hard_comp=0.100000:first_pts=0" -bsf:v h264_mp4toannexb -f mpegts "rtmp://<my-ip>/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream live=1"

I am getting the following :-

    rtmp @ 0x5ad3bda4fe00] Detected librtmp style URL parameters, these aren't supported by the libavformat internal RTMP handler 
    currently enabled. See the documentation for the correct way to pass parameters.
    [tcp @ 0x5ad3bda4f680] Connection to tcp://81.174.241.153:1935 failed: Connection refused
    [rtmp @ 0x5ad3bda4fe00] Cannot open connection tcp://81.174.241.153:1935
    rtmp://81.174.241.153/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream 
    live=1: Connection refused

I have `sudo ufw allow 1935/tcp`

**Comment by AaronNGray** [comment]
Okay I opened port 1935 on my router and ffmpeg is streaming now. But restream is not seeing it still.

I have the `rtmp://<my-static-ip>/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream` from restream and :-

    ffmpeg -re -i "big_buck_bunny_720p_h264.mov" -codec copy -f flv -listen 1 "rtmp://<my-static-ip>/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream"
    
But ffmpeg is streaming then comes up with :-

    av_interleaved_write_frame(): Broken pipe
    Last message repeated 1 times
    Error writing trailer of rtmp://<my-static-ip>/live/bbb/416a8d4d-a226-4a4f-aa60-301cf7d51b15.stream/817b9369-0be5-40c5-92dd-9ef78f4e4a7f.stream: Broken pipe

**Comment by AaronNGray** [comment]
okay removed `-listen 1` and its working !!!
